barman/storage/file_stats.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# © Copyright EnterpriseDB UK Limited 2013-2021
+#
+# This file is part of Barman.
+#
+# Barman is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Barman is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Barman.  If not, see <http://www.gnu.org/licenses/>.
+
+from datetime import datetime
+
+try:
+    from datetime import timezone
+
+    utc = timezone.utc
+except ImportError:
+    # python 2.7 compatibility
+    from dateutil import tz
+
+    utc = tz.UTC
+
+
+class FileStats:
+    def __init__(self, size, last_modified):
+        """
+        Arbitrary timezone set to UTC. There is probably possible improvement here.
+        :param size: file size in bytes
+        :type size: int
+        :param last_modified: Time of last modification in seconds
+        :type last_modified: int
+        """
+        self.size = size
+        self.last_modified = datetime.fromtimestamp(last_modified, tz=utc)
```

Comment:
My immediate thought here was "we should use whatever time zone pg_basebackup uses which will surely be the time zone of the PostgreSQL server" however a few tests with PG14 showed that the backup manifest produced by pg_basebackup would use GMT regardless of the PostgreSQL time zone or the OS time zone.

The PG docs don't specify the time zone but [this mailing list discussion](https://postgrespro.com/list/id/4cea0d44-7524-dc7d-9043-8c701ccaba0a@oss.nttdata.com) suggests Last-Modified is always GMT.

Reply:
I indeed noticed that GMT is used in backup manifest. I tested pg_verifybackup with random srting, it is not checked and just informational.
For consistency, I should probably add GMT to formated date.